### PR TITLE
[JW8-10728] Typescript 3 files

### PR DESCRIPTION
**Pull request by pajong**
### This PR will...
- Typescript the following 3 files:
-- `custom-button.js`
-- `simple-tooltip.js`
-- `volumetooltipicon.js`

#### Addresses Issue(s):

JW8-10728

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: src/js/view/controls/components/volumetooltipicon.ts [hunk 1]
```diff
@@ -74,38 +84,37 @@ export default class VolumeTooltipIcon extends TooltipIcon {
         this.onAudioMode(null, _model.get('audioMode'));
 
         this._model.on('change:audioMode', this.onAudioMode, this);
-        this._model.on('change:volume', this.onVolume, this);
```
Comment (pajong) [inline src/js/view/controls/components/volumetooltipicon.ts:77]: `onVolume` is not defined in this class, so removing this line

**Comment by jwrobot-player** [comment]
[:exclamation:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1636/) MULTI Build for commit 9e040ed931a4e75b29e1531b2693ed18972f9f0f did not complete (FAILURE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/none/consoleFull) jwplayer build NOT_STARTED
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6003/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6003/consoleFull) jwplayer unit tests FAILURE
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/consoleFull) jwplayer-commercial build NOT_STARTED
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/none/consoleFull) jwplayer-commercial unit tests NOT_STARTED
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/none/allure/) NOT_STARTED
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/view/) Views

**Comment by jwrobot-player** [comment]
[:exclamation:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1638/) MULTI Build for commit 9e040ed931a4e75b29e1531b2693ed18972f9f0f did not complete (FAILURE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/8966/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/8966/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6005/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6005/consoleFull) jwplayer unit tests FAILURE
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/consoleFull) jwplayer-commercial build NOT_STARTED
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/none/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/none/consoleFull) jwplayer-commercial unit tests NOT_STARTED
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/none/allure/) NOT_STARTED
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/none/archive/test/view/) Views

**Comment by jwrobot-player** [comment]
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1639/) MULTI Build for commit 40416014997435211cb672dc771dd2f85edbc7d5 is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/8967/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/8967/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6006/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6006/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/27117/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/27117/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/22742/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/22742/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/5000/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/27117/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/27117/archive/test/view/) Views

File: src/js/view/controls/components/custom-button.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import svgParse from 'utils/svgParser';
 
 let collection = {};
 
-export function getCachedIcon(svg) {
+export function getCachedIcon(svg: string): HTMLElement {
```
Comment (robwalch) [inline src/js/view/controls/components/custom-button.ts:7]: How about typing `collection`?

The [Record](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype) utility type is a great way to do this (especially if you want to type the svg name strings):

```ts
	let collection: Record<string, XMLDocument> = {};
```

or as the messier key-value pair:
```ts
	let collection: {[svg: string]: XMLDocument} = {};
```

**Comment by jwrobot-player** [comment]
[:warning:](https://jenkins-player.longtailvideo.com/job/jw7-pr-multi-opensource/1640/) MULTI Build for commit 39ed6f4d81b9e39c48c18e52b5967dd3930acadd is unstable (UNSTABLE).
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/8968/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource/8968/consoleFull) jwplayer build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6007/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-opensource-unit-tests/6007/consoleFull) jwplayer unit tests SUCCESS
[:building_construction:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/27125/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/27125/consoleFull) jwplayer-commercial build SUCCESS
[:clipboard:](https://jenkins-player.longtailvideo.com/job/jw7-commercial/22747/)[:black_small_square:](https://jenkins-player.longtailvideo.com/job/jw7-commercial-unit-tests/22747/consoleFull) jwplayer-commercial unit tests SUCCESS
[:cucumber: Cucumber Integration Tests](https://jenkins-player.longtailvideo.com/job/jw7-commercial-cucumber-wdio/5008/allure/) UNSTABLE
[:jack_o_lantern:](http://player-pr-test-jenkins.longtailvideo.com/builds/27125/archive/test/squash/) Manual Tests
[:tv:](http://player-pr-test-jenkins.longtailvideo.com/builds/27125/archive/test/view/) Views

**Review by robwalch — APPROVED** [review]

**Pull request merged**